packages/@sanity/client/sanityClient.d.ts

```diff
@@ -0,0 +1,1784 @@
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: FirstDocumentMutationOptions): Promise<SanityDocument>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to an array of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: AllDocumentsMutationOptions): Promise<SanityDocument[]>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the first mutated document
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: FirstDocumentIdMutationOptions): Promise<string>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options: AllDocumentIdsMutationOptions): Promise<string[]>
+
+  /**
+   * Commit the transaction, returning a promise that resolves to the document ID of the mutated documents
+   *
+   * @param options Options for the mutation operation
+   */
+  commit(options?: BaseMutationOptions): Promise<string[]>
+
+  /**
+   * Clears the transaction of all operations
+   */
+  reset(): this
+}
+
+interface BaseClientConfig {
+  useCdn?: boolean
+  token?: string
+  apiHost?: string
+  ignoreBrowserTokenWarning?: boolean
+  withCredentials?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  gradientMode?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  namespace?: string
+
+  /**
+   * @deprecated Don't use
+   */
+  useProjectHostname?: boolean
+
+  /**
+   * @deprecated Don't use
+   */
+  requester?: GetItRequester
+}
+
+type ProjectlessClientConfig = BaseClientConfig & {
+  useProjectHostname: false
+  projectId?: string
+  dataset?: string
+}
+
+type ProjectClientConfig = {
+  projectId: string
+  dataset?: string
+} & BaseClientConfig
+
+type ClientConfig = ProjectClientConfig | ProjectlessClientConfig
+
+interface RequestOptions {
+  timeout?: number
+  token?: string
+}
+
+type BaseMutationOptions = RequestOptions & {
+  visibility?: 'sync' | 'async' | 'defer'
+  returnDocuments?: boolean
+  returnFirst?: boolean
+}
+
+type MutationEvent = {
+  type: 'mutation'
+  documentId: string
+  eventId: string
+  identity: string
+  mutations: Mutation[]
+  previousRev?: string
+  resultRev?: string
+  result?: SanityDocument
+  previous?: SanityDocument | null
+  timestamp: string
+  transactionId: string
+  transition: 'update' | 'appear' | 'disappear'
+}
+
+type ChannelErrorEvent = {
+  type: 'channelError'
+  message: string
+}
+
+type DisconnectEvent = {
+  type: 'disconnect'
+  reason: string
+}
+
+type ReconnectEvent = {
+  type: 'reconnect'
+}
+
+type WelcomeEvent = {
+  type: 'welcome'
+}
+
+type ListenEvent =
+  | MutationEvent
+  | ChannelErrorEvent
+  | DisconnectEvent
+  | ReconnectEvent
+  | WelcomeEvent
+
+interface ListenOptions {
```

Comment:
I have no idea of the usage scope for all of the interfaces, but the ones that play a role as inputs, those should definitely be exported.

Use cases: decorators.

In this case, I have custom React Hooks that wrap around Sanity.  
Before trying out these - official TS defs, I had my own that exposed those that I found worthy.  
Since I am making them as generics (looking forward to contribute to OSS community), they expose full API surface, in `useSubscription` case, allowing to set custom `ListenOptions` per hook-usage.

Reply:
Agreed, and I see your point. I've exported a bunch of interfaces/types now.